### ISSUE TITLE
RCS1163 should not report method that is referenced as method group

### ISSUE THREAD

**Issue by JosefPihrt** [JosefPihrt/Roslynator#51]

**Comment by jnyrup** [comment]
Does this generate too many false positives?


We fixed bug today, which I guess could have been caught by RCS1163:
```csharp
var first = list.First();
list.Select(e => first.Property); // <-- 'e' is unused, but should have been used instead of 'first'
```

I see that RCS1163 is triggered for local functions:
```csharp
var local = new int[] { 1 }.Select(localSelector).ToArray();
int localSelector(int a) => 0; // triggers RCS1163 on 'a'
```

**Comment by JosefPihrt** [comment]
RCS1163 does not analyze parameters of lambda expression.

But it probably could.

**Comment by JosefPihrt** [comment]
@jnyrup Please file a new issue if you are interested in such functionality.
